app/models/server/models/LivechatRooms.js

```diff
@@ -333,6 +349,16 @@ export class LivechatRooms extends Base {
 		return this.find(query);
 	}
 
+	findOpenByVisitorId(visitorId) {
+		const query = {
+			't': 'l',
+			'v._id': visitorId,
```

Comment:
not sure if that would help, but is it possible to change to use `v.token` instead of `v._id`? trying to use the best index here =)

Reply:
Good catch, I reused one function we already had, but we can use another that fetch using token